src/jit/compiler.cpp

```diff
@@ -1007,6 +1008,7 @@ var_types Compiler::getReturnTypeForStruct(CORINFO_CLASS_HANDLE clsHnd,
         // and also examine the clsHnd to see if it is an HFA of count one
         useType = getPrimitiveTypeForStruct(structSize, clsHnd);
     }
+#endif
```

Comment:
As I mentioned here(https://github.com/dotnet/coreclr/issues/10027#issuecomment-285281093), 

P/Invoke is done like:
```
(Managed Code) --(1)--> (IL_STUB_PInvoke) --(2)--> (Native Code)
```

I think it(return by EAX) is the convention for (1). We actually need return buffer for (2).
Now I see this again, this change is probably unnecessary since it will change the calling convention for all managed code.